block/retriever.go

```diff
@@ -47,6 +47,12 @@ func (m *Manager) ApplyBatchFromSL(slBatch *settlement.Batch) error {
 			m.blockCache.Delete(block.Header.Height)
 		}
 	}
+
+	// validate the batch applied successfully and we are at the end height
```

Comment:
if the batch isn't applied succesfully wouldn't we get an err from the `applyBlockWithFraudHandling` ?

Reply:
we're looping over `for i, block := range batch.Blocks`, which data on the DA
it can be empty/missing